src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -36,7 +36,24 @@ public Task<Geoposition> GetGeopositionAsync()
 		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
 			=> GetGeopositionAsync();
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
```

Comment:
What is the behavior of `RequestAccessAsync()` on UWP ? does it prompt the user in some way? Because I believe the code `ContextCompat.CheckSelfPermission` only checks if the permission is granded, but doesn't prompt the user. Basing this comment off this [link](https://developer.android.com/reference/android/support/v4/content/ContextCompat.html#checkSelfPermission(android.content.Context,%20java.lang.String))

Reply:
If UWP prompts the user, we should also prompt on Android. 

If that's the case, this code could be good inspiration! rootActivityProvider could be ContextHelper.Current.
If there's a way it would be nice to request the proper permission (Coarse or Fine) based on what is in the manifest. If Fine is in the manifest request Fine, otherwise request Coarse.
```		public Task<bool> TryGetPermission(CancellationToken ct, string permissionIdentifier) => _tryGetPermission(ct, permissionIdentifier);

		private async Task<bool> TryGetPermissionCore(CancellationToken ct, string permissionIdentifier)
		{
			// 1. ActivityCompat.RequestPermissions is a non-cancellable operation.
			// 2. Requesting 2 permissions at the same time is invalid on Android;
			//    the requests must either be serialized or aggregated.
			//  To avoid this, this task is shared (LockInvocation(InvocationLockingMode.Share))
			//  which means any pending task will be used instead of creating a new one.
			//  This creates a problem where cancelling the source task would also
			//  cancel the operation for other callers which is invalid.
			ct = CancellationToken.None;

			if (this.Log().IsEnabled(LogLevel.Debug))
			{
				this.Log().Debug($"Trying to get permission '{permissionIdentifier}'.");
			}

			var rootActivity = await _rootActivityProvider(ct);

			if (ActivityCompat.CheckSelfPermission(rootActivity, permissionIdentifier) == Permission.Granted)
			{
				if (this.Log().IsEnabled(LogLevel.Information))
				{
					this.Log().Info($"Permission '{permissionIdentifier}' granted.");
				}

				return true;
			}

			var code = Interlocked.Increment(ref _permissionRequest);

			var onRequestPermissionsResultObservable = await _onRequestPermissionsResultObservableProvider(ct);

			var permissionTask = onRequestPermissionsResultObservable
				.Where(r => r.RequestCode == code)
				.FirstAsync(ct);

			ActivityCompat.RequestPermissions(rootActivity, new[] { permissionIdentifier }, code);

			var result = await permissionTask;

			bool granted = result.GrantResults[0] == Permission.Granted;

			if (this.Log().IsEnabled(LogLevel.Information))
			{
				this.Log().Info($"Permission '{permissionIdentifier}' {(granted ? "granted" : "denied")}.");
			}

			return granted;
		}
```

Reply:
Yes, UWP displays window asking user for permission (if no permission is granted).
But I cannot emulate this window (my Android knowledge is 'too short'), and helper method (PermissionsHelper.CheckFineLocationPermission, in src\Uno.UI\Extensions\PermissionsHelper.Android.cs / src\Uno.UWP\Extensions\PermissionsHelper.cs) doesn't work.

Returning Granted always is worse than returning Granted/Denied after checking permission.

Yes, Uno.Geolocation for Android should be deeply rewritten; I made it almost work in my app; and this PR is only a start. But as I made Uno.Geolocation working on two devices only (my tablet, and @zipswitch device), rest of changes should wait.

My PC is too old to build Uno on it. To build Uno, I use Azure VM (provided by Uno team).
So, I cannot trace (step by step) what happens, as VM cannot start android emulator, and my PC cannot trace into Uno :)